protocol.md

```diff
@@ -568,6 +568,127 @@ Tus-Resumable: 1.0.0
 Upload-Offset: 11
 ```
 
+### Partial Checksum
+
+This extension allows Clients that use the Checksum extension to verify and confirm the
+integrity of chunks that were not received in full by the Server - thereby avoiding that
+intact data needs to be sent again.
+
+Servers that support this extension MUST add `partial-checksum` to the `Tus-Extension` header.
+
+If a Server receives an incomplete `PATCH` or `POST` request with `Upload-Checksum` header, it
+MAY store the incompletely received chunk's data (hereafter referred to as *partial data*) in
+a separate location instead of discarding it.
+
+If the Server receives a `HEAD` request on the upload thereafter, it MAY compute a checksum
+on the partial data and then provide the result as well as the range of the partial data within
+the upload through the `Upload-Partial-Checksum` and `Upload-Partial-Checksum-Range`
+response headers.
+
+A Client can then use the range returned by the Server, compute its own checksum for the 
+same range and compare it against the checksum returned by the Server.
+
+If the checksums are identical, the Client SHOULD confirm the partial data's integrity to the
+Server by sending the checksum and range in the `Upload-Partial-Checksum` and
+`Upload-Partial-Checksum-Range` headers of the next `PATCH` request. The `Upload-Offset`
+header of the `PATCH` request MUST take the confirmed range into account and represent
+the actual offset from where the upload should be resumed.
+
+If the checksums don't match, the Client MUST continue the upload from the `Upload-Offset`
+returned by the `HEAD` request, at which point the Server MAY discard any partial data it may
+still be holding.
+
+If a Server receives  `Upload-Partial-Checksum` and `Upload-Partial-Checksum-Range`
+headers as part of a `PATCH` request and they match its own values for partial data stored
+for the upload, the Server MUST append it to the upload and move its internal upload offset
+forward accordingly, before processing the request any further.
```

Comment:
I find this part a bit hard to read. Basically what you are saying is that if the `Upload-Partial-Checksum` header matches for the range specified in `Upload-Partial-Checksum-Range` then the server should no longer count this as partial data but instead as part of the file (either by copying it to the actual file or removing a marker or similar)?

Reply:
Yes, exactly that.

I tried to make it a bit easier to read and arrived at this (in the next commit):

> If a Server receives `Upload-Partial-Checksum` and `Upload-Partial-Checksum-Range`
> headers as part of a `PATCH` request and they match the Server's computed results for its stored
> partial data for the upload, the Server MUST append the partial data to the upload and move its
> internal upload offset forward accordingly, before processing the request any further.

Reply:
Much better, thank you.